### PR TITLE
feat: added BOX doctype, Shipping Information as child table in Item

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1199520543546492)

### CODE REVIEW

File: erpnext/stock/doctype/item/item.js
Comment (hrwX) [inline erpnext/stock/doctype/item/item.js]:
```suggestion
		let row = locals[cdt][cdn];
```

File: erpnext/stock/doctype/item/item.js
Comment (hrwX) [inline erpnext/stock/doctype/item/item.js]:
```suggestion
frm.set_query('shipping_information', 'uom', function(doc,cdt,cdn) {
```

Move this to refesh trigger

File: erpnext/stock/doctype/item/item.js
Comment (hrwX) [inline erpnext/stock/doctype/item/item.js]:
```suggestion
	let row = locals[cdt][cdn];
```

File: erpnext/stock/doctype/item/item.js
Comment (hrwX) [inline erpnext/stock/doctype/item/item.js]:
```suggestion
	let uom = doc.shipping_information.map(info => info.uom);
	let box = doc.shipping_information.map(info => info.box);
```

File: erpnext/stock/doctype/item/item.js
Comment (hrwX) [inline erpnext/stock/doctype/item/item.js]:
```suggestion
	if (box.includes(row.box)) {
```

File: erpnext/stock/doctype/item/item.js
Comment (hrwX) [inline erpnext/stock/doctype/item/item.js]:
```suggestion
			filters : {
```

File: erpnext/stock/doctype/packing_slip/packing_slip.js
Comment (hrwX) [inline erpnext/stock/doctype/packing_slip/packing_slip.js]:
```suggestion
		let row = locals[cdt][cdn];
```

File: erpnext/stock/doctype/packing_slip/packing_slip.js
Comment (hrwX) [inline erpnext/stock/doctype/packing_slip/packing_slip.js]:
```suggestion
					if (r && r.message) {
```

File: erpnext/stock/doctype/packing_slip/packing_slip.js
Comment (hrwX) [inline erpnext/stock/doctype/packing_slip/packing_slip.js]:
```suggestion
						frappe.model.set_value(cdt, cdn, "no_of_boxes_required", r.message);
```

File: erpnext/stock/doctype/packing_slip/packing_slip.js
Comment (hrwX) [inline erpnext/stock/doctype/packing_slip/packing_slip.js]:
```suggestion
			});
```

File: erpnext/stock/doctype/packing_slip/packing_slip.js
Comment (hrwX) [inline erpnext/stock/doctype/packing_slip/packing_slip.js]: Change this to one suggested above

File: erpnext/stock/doctype/packing_slip/packing_slip.py
Comment (hrwX) [inline erpnext/stock/doctype/packing_slip/packing_slip.py]: Use ORM

**Review by hrwX — CHANGES_REQUESTED** [review]

**Comment by rvpasari** [comment]
Waiting on a sign off on the functionality from Kenneth before I approve the PR.

**Review by rvpasari — DISMISSED** [review]

**Review by rvpasari — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**
